### PR TITLE
wooting: update udev rules

### PR DESCRIPTION
**Pull request by jtrees**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

Related: #225053

### CODE REVIEW

**Comment by Jayman2000** [comment]
I can vouch for this and #225053. I’m using an overlay to replace the `wooting-udev-rules` package with the one from this PR and the `wootility` package with the one from the other PR. That overlay allows me to use the Wootility with my Wooting Two HE.

File: pkgs/os-specific/linux/wooting-udev-rules/default.nix [hunk 1]
```diff
@@ -2,9 +2,9 @@
 
 stdenv.mkDerivation rec {
   pname = "wooting-udev-rules";
-  version = "20210525";
+  version = "20230331";
```
Comment (SuperSandro2000) [inline pkgs/os-specific/linux/wooting-udev-rules/default.nix:5]:
```suggestion
  version = "unstable-2023-03-31";
```

File: nixos/modules/hardware/wooting.nix [hunk 2]
```diff
@@ -2,8 +2,15 @@
 
 with lib;
 {
-  options.hardware.wooting.enable =
-    mkEnableOption (lib.mdDoc "support for Wooting keyboards");
+  options.hardware.wooting.enable = mkOption {
```
Comment (SuperSandro2000) [inline nixos/modules/hardware/wooting.nix:5]: Please revert this and we also don't want to activate this by default

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/os-specific/linux/wooting-udev-rules/default.nix (hunk 1 above)
Comment (jtrees) [reply pkgs/os-specific/linux/wooting-udev-rules/default.nix:5]: Putting "unstable" in there kinda implies there is a stable version, no?

But it's fine. I changed it.

File: nixos/modules/hardware/wooting.nix (hunk 2 above)
Comment (jtrees) [reply nixos/modules/hardware/wooting.nix:5]: My intention here was to add the following clarification:

> Note that users must be in the "input" group for udev rules to apply.

Is there a better way to do this?

Comment (jtrees) [reply nixos/modules/hardware/wooting.nix:5]: Good point about the default. I changed it to disabled.

**Comment by jtrees** [comment]
@SuperSandro2000 Thanks for your review.

File: pkgs/os-specific/linux/wooting-udev-rules/default.nix (hunk 1 above)
Comment (SuperSandro2000) [reply pkgs/os-specific/linux/wooting-udev-rules/default.nix]: unstable means there is no tagged version upstream in nixpkgs

File: nixos/modules/hardware/wooting.nix (hunk 2 above)
Comment (SuperSandro2000) [reply nixos/modules/hardware/wooting.nix:5]: > > Note that users must be in the "input" group for udev rules to apply.

that can be done be overwriting the description

File: pkgs/os-specific/linux/wooting-udev-rules/default.nix [hunk 3]
```diff
@@ -2,9 +2,9 @@
 
 stdenv.mkDerivation rec {
   pname = "wooting-udev-rules";
-  version = "20210525";
+  version = "unstable-20230331";
```
Comment (SuperSandro2000) [inline pkgs/os-specific/linux/wooting-udev-rules/default.nix:5]:
```suggestion
  version = "unstable-2023-03-31";
```
for dates we use iso format

File: nixos/modules/hardware/wooting.nix [hunk 4]
```diff
@@ -2,8 +2,15 @@
...
+    type = types.bool;
+    default = false;
+    description = lib.mdDoc ''
+      Whether to enable support for Wooting keyboards.
+
+      Note that users must be in the "input" group for udev rules to apply.
+    '';
```
Comment (SuperSandro2000) [inline nixos/modules/hardware/wooting.nix:12]:
```suggestion
  options.hardware.wooting.enable = mkEnableOption (lib.mdDoc ''support for Wooting keyboards.
    Note that users must be in the "input" group for udev rules to apply'');
```

File: nixos/modules/hardware/wooting.nix (hunk 2 above)
Comment (jtrees) [reply nixos/modules/hardware/wooting.nix:5]: Done.

File: nixos/modules/hardware/wooting.nix (hunk 4 above)
Comment (jtrees) [reply nixos/modules/hardware/wooting.nix:12]: Thanks. Updated.

File: pkgs/os-specific/linux/wooting-udev-rules/default.nix (hunk 3 above)
Comment (jtrees) [reply pkgs/os-specific/linux/wooting-udev-rules/default.nix:5]: Ah yes. I missed that first time round. Fixed.

**Pull request merged**